File: step14/jeongkpa/15649.c
# N과 M

N, M = list(map(int,input().split()))

M_list = []

def func():
  if len(M_list) == M: #M_list 안에 리스트 갯수가 M개 인경우 (재귀 종료 조건)
    print(' '.join(map(str,M_list))) # 그때 안에 있는 M_list 출력
    return

  for i in range(1,N+1):
    if i in M_list:
        continue
    M_list.append(i)
    func()
    M_list.pop()

func()

File: 백트래킹/15649.py
#[15649]백트래킹, N과 M(1)
#1부터 N까지 자연수 중에서 중복 없이 길이가 M개인 수열 출력
#백트래킹(Backtracking, 퇴각검색)

n, m = map(int, input().split())

s = []
def f():
  if len(s) == m:
    print(' '.join(map(str, s)))
    return

  for i in range(1, n + 1):
    if i in s: #이미 선택한 숫자를 다시 선택하려 하면 배제
      continue
    s.append(i) #스택에 추가
    f() #동작
    s.pop() #다시 스택에서 빼내기

f()